### DISCUSSION TITLE
[Poll] Catalog Items API - All Attributes or Editable Attributes?

### DISCUSSION THREAD

**Discussion by chapmanjw** [amzn/selling-partner-api-models#604] [Ideas]
**Question**

Would you prefer the Catalog Items API to continue returning only editable attributes or return all known attributes, even if they cannot be edited yet in the Listings Items APIs?

Use the following reactions to answer:

- ❤️  Editable Attributes Only (Current Behavior)
- 🚀  All Known Attributes 

**Background**

When `attributes` are requested in the [Catalog Items API](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/use-case-guides/catalog-items-api-use-case-guide/catalog-items-api-use-case-guide_2020-12-01.md), they are currently limited to the attributes available in the [Product Type Definitions API](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/use-case-guides/product-type-definitions-api-use-case-guide/definitions-product-types-api-use-case-guide_2020-09-01.md). For product types not yet launched in the Product Type Definitions API, the attributes are limited to the attributes defined on the root `PRODUCT` product type. The root `PRODUCT` product type only defines attributes common to all product types and does not include attributes that only apply to some product types. 

As Amazon is currently underway migrating and launching new product type definitions, this means that for the majority of products that are retrieved from the Catalog Items API only attributes defined on the `PRODUCT` product type will be returned. Examples of attributes that are not returned when the product type of an ASIN is not launched, such as `BEAUTY`, include `unit_count`, `material_composition`, `hair_type`, `safety_warning`, and `scent`.

We are gathering feedback from developers on which approach is preferable for the Catalog Items API: 

- ❤️  Continue returning only attributes defined by the Product Type Definitions API and editable in the Listings Items API
- 🚀  Return all known attributes regardless if they are defined by the Product Type Definitions API or can be edited in the Listings Items API.

Thanks!

**Comment by geoffrey-b-pacvue** [comment]
All consumer facing attributes should be editable to ensure that amazon customers get the right information regarding pack size, color of products....

**Reply by chapmanjw** [reply]
@geoffrey-b-pacvue Yes, agreed. That is dependent on the migration and launch of product types in the Product Types Definitions APIs and Listings APIs (for SKU data). Until that migration is complete, we are considering options on which attributes to provide in the Catalog Items APIs (the ASIN data).

**Reply by geoffrey-b-pacvue** [reply]
One additional element to add to this list of attributes is the "availability status" of a product available under item > catalog. Let me know if this attribute has the same dependencies that you called out.

**Comment by sbosman1** [comment]
List attributes that are required are optional attributes. So per attribute need definition if its required or not.
Complete list of categories including mapping with producttype that are possible on the marketplace. So we can use the complete API for integration instead of different sources like the excel BTG. 

This will make the API a full API.

**Comment by m8labsteam** [comment]
Upvoted very much needed. Like Geoffrey "says customers get the right information regarding pack size, color of products...." The new API should be an improvement, not a step backward.

**Comment by booktrakker** [comment]
We need visibility into all data irrespective of whether it can be edited, and in our case, as you probably know by now, we need Author.  We cannot complete our Migration to SP-API if we cannot get Author data as our sellers are booksellers, and Author is about as critical a piece of data as there can be.

Please get this resolved as soon as possible.  We cannot meet the deadlines if we cannot release the new code.

**Reply by bigindian66** [reply]
Same here, we're working with books and need the author name at least !

**Reply by booktrakker** [reply]
**Can we get an ETA for when Author will be included?**  We cannot release until it is, and Amazon wants to impose deadlines, so we really need to understand what to expect.
I had to revert much of our code back to MWS because of this and other unresolved issues.  We cannot release if the new API does not function at least as well as the old one.

**Comment by cmlkx** [comment]
@chapmanjw  @sungolivia  All known attributes please!!!  If you want to tag them as editable or not that would be a "nice to have" feature but for us we would 100% prefer visibility to all item attributes

**Comment by gokigoks** [comment]
We are looking for Artist, Release date, and Label, among a few others. It would be nice to have all the fields previously offered by MWS. Hopefully, these get added soon.

**Comment by cmlkx** [comment]
Same situation as @gokigoks, we need all the productgroup details that were present on CatalogsAPI v0 and MWS.  @sungolivia  @chapmanjw  any updates on this?

**Comment by cmlkx** [comment]
@sungolivia @chapmanjw  Just got the 30 day notice.  We would apricate some kind of update on this issue.
